src/Microsoft.DotNet.Darc/src/Darc/Operations/AddBuildToChannelOperation.cs

```diff
@@ -186,7 +186,16 @@ private async Task<int> PromoteBuildAsync(Build build, Channel targetChannel, IR
         /// <param name="build">Build for which the Arcade SDK dependency build will be inferred.</param>
         private async Task<(string sourceBranch, string sourceVersion)> GetSourceBranchInfoAsync(Build build)
         {
-            if (!string.IsNullOrEmpty(_options.SourceBranch) && !string.IsNullOrEmpty(_options.SourceSHA))
+            var hasSourceBranch = !string.IsNullOrEmpty(_options.SourceBranch);
+            var hasSourceSHA = !string.IsNullOrEmpty(_options.SourceSHA);
+
+            if (hasSourceBranch ^ hasSourceSHA)
+            {
+                Console.WriteLine("The `source-branch` and `source-sha` parameters need to be specified together.");
+                return (null, null);
+            }
+
+            if (hasSourceBranch && hasSourceSHA)
```

Comment:
Does the AzDO API still work if you just specify the source branch?

Reply:
Yes, it can. I'll patch the PR to let that happen.

Reply:
Done.

Reply:
PTAL